### PR TITLE
message pass to dpif_netlink_class->tt_table_add successed.

### PR DESCRIPTION
**Pull request by zhanghaooo**

### CODE REVIEW

File: lib/dpif-netdev.c [hunk 1]
```diff
@@ -3888,6 +3888,7 @@ const struct dpif_class dpif_netdev_class = {
     NULL,                       /* ct_dump_next */
     NULL,                       /* ct_dump_done */
     NULL,                       /* ct_flush */
+	NULL,                       /* tt_table_add */
```
Comment (chenwhql) [inline lib/dpif-netdev.c]: pay attention to alignment

File: lib/dpif-netlink.c [hunk 2]
```diff
@@ -2342,6 +2342,33 @@ dpif_netlink_ct_flush(struct dpif *dpif OVS_UNUSED, const uint16_t *zone)
 }
 #endif
 
+static int
+dpif_netlink_tt_table_add(struct dpif_tt_flow *tt_flow)
+{
+	VLOG_WARN("Pass dpif to netlink success!");
+	tt_flow->port = 0;
+	/*
+    struct dpif_netlink_tt tt_request, tt;
```
Comment (chenwhql) [inline lib/dpif-netlink.c]: alignment problem, same as the last one

File: lib/ofp-util.c [hunk 3]
```diff
@@ -9841,3 +9841,31 @@ ofputil_encode_get_async_config(const struct ofp_header *oh,
...
+
+
+enum ofperr 
+ofputil_decode_tt_table_mod(const struct ofp_header *oh,
+							struct ofputil_tt_table_mod *ttm)
+{
+	struct onf_tt_flow_mod *tx_ttm;
+	struct ofpbuf msg;
+    
+    /* get the struct tx_tt_table_mod in the openflow message
```
Comment (yearsj) [inline lib/ofp-util.c:9854]: 空格没有对齐

File: lib/ofp-util.h [hunk 4]
```diff
@@ -1328,4 +1329,20 @@ enum ofperr ofputil_decode_requestforward(const struct ofp_header *,
...
+    uint8_t port; /* The entry related port. */
+    uint8_t etype; /* Send entry or receive entry. */
+    uint8_t flow_id; /* The identify of a flow. */
+    ovs_be32 scheduled_time; /* The scheduled time that the flow packet is received or sent. */
+    ovs_be32 period; /* The scheduling period. */
+    ovs_be32 buffer_id; /* Buffered packet to apply to. */
+    ovs_be32 pkt_size; /* The flow packet size. */
+};
+
+enum ofperr ofputil_decode_tt_table_mod(const struct ofp_header *oh,
```
Comment (yearsj) [inline lib/ofp-util.h:1356]: 格式问题

File: lib/ofp-util.c [hunk 5]
```diff
@@ -9841,3 +9841,31 @@ ofputil_encode_get_async_config(const struct ofp_header *oh,
...
+    ofpbuf_use_const(&msg, oh, ntohs(oh->length));
+    ofpraw_pull_assert(&msg);
+	
+    tx_ttm = ofpbuf_pull(&msg, sizeof *tx_ttm);
+    /* tx_tt_table_mod --> ofputil_tt_table_mod */
+    ttm->command = tx_ttm->command;
+    ttm->port = tx_ttm->port;
+    ttm->etype = tx_ttm->etype;
+    ttm->flow_id = tx_ttm->flow_id;
+    ttm->scheduled_time = ntohs(tx_ttm->scheduled_time);
```
Comment (chenwhql) [inline lib/ofp-util.c:9865]: the 4 members here are 32-bit variable, maybe need to use `ntohl` function

File: ofproto/ofproto-dpif.c [hunk 6]
```diff
@@ -5753,6 +5753,23 @@ ofproto_dpif_delete_internal_flow(struct ofproto_dpif *ofproto,
...
+static enum ofperr 
+tt_table_add(const struct ofproto *ofproto_, const struct ofputil_tt_table_mod *mod)
+{
+    struct ofproto_dpif *ofproto = ofproto_dpif_cast(ofproto_);
+    struct dpif_tt_flow tt_flow;
+    tt_flow.command = mod->command;
+    tt_flow.port = mod->port;
+    tt_flow.etype = mod->etype;
+    tt_flow.flow_id = mod->flow_id;
+    tt_flow.scheduled_time =mod->scheduled_time;
```
Comment (chenwhql) [inline ofproto/ofproto-dpif.c]: lack of a space after equal

File: lib/ofp-util.c (hunk 3 above)
Comment (zhanghaooo) [reply lib/ofp-util.c:9854]: done

File: lib/ofp-util.h (hunk 4 above)
Comment (zhanghaooo) [reply lib/ofp-util.h:1356]: done

**Review by chenwhql — APPROVED** [review]

**Pull request merged**